src/ToggleSwitch.tsx

```diff
@@ -0,0 +1,329 @@
+import React, {MouseEventHandler, useCallback, useEffect} from 'react'
+import styled, {css} from 'styled-components'
+import {variant} from 'styled-system'
+import {Box, Spinner, Text} from '.'
+import {get} from './constants'
+import {useProvidedStateOrCreate} from './hooks'
+import sx, {BetterSystemStyleObject, SxProp} from './sx'
+import VisuallyHidden from './_VisuallyHidden'
+
+const TRANSITION_DURATION = '80ms'
+const EASE_OUT_QUAD_CURVE = 'cubic-bezier(0.5, 1, 0.89, 1)'
+
+type SwitchProps = {
+  /** The id of the DOM node that describes the switch */
+  ['aria-describedby']?: string
+  /** The id of the DOM node that labels the switch */
+  ['aria-labelledby']: string
+  /** Uncontrolled - whether the switch is turned on */
+  defaultChecked?: boolean
+  /** Whether the switch is ready for user input */
+  disabled?: boolean
+  /** Whether the switch's value is being calculated */
+  loading?: boolean
+  /** Whether the switch is turned on */
+  checked?: boolean
+  /** The callback that is called when the switch is toggled on or off */
+  onChange?: (on: boolean) => void
+  /** The callback that is called when the switch is clicked */
+  onClick?: MouseEventHandler
+  /** Size of the switch */
+  size?: 'small' | 'medium'
+  /** Whether the "on" and "off" labels should appear before or after the switch.
+   * **This should only be changed when the switch's alignment needs to be adjusted.** For example: It needs to be left-aligned because the label appears above it and the caption appears below it.
+   */
+  statusLabelPosition?: 'start' | 'end'
+} & SxProp
+
+const sizeVariants = variant({
+  prop: 'size',
+  variants: {
+    small: {
+      height: '24px',
+      width: '48px'
+    }
+  }
+})
+
+type SwitchButtonProps = {
+  disabled?: boolean
+  checked?: boolean
+  size?: SwitchProps['size']
+} & SxProp
+
+type InnerIconProps = {size?: SwitchProps['size']}
+
+const CircleIcon: React.FC<InnerIconProps> = ({size}) => (
```

Comment:
Should these be added to Octicons? cc @ashygee @edokoa

Reply:
I didn't include them because I didn't think we'd ever want to use them in isolation. More than happy to add them if you and other folks think we should.